### PR TITLE
Add NPM as a dependency when building from source + Troubleshooting instructions

### PR DESCRIPTION
**Pull request by silentninja**
Addresses concerns [raised by the tester](https://matrix.to/#/!dwNKVEeYPiAqvkytob:matrix.mathesar.org/$QcgkedPKa5NVGN9BsRYtlMrRznQ3hPv0huJlfXMtRd8?via=matrix.mathesar.org&via=matrix.org) to mention `npm` as a dependency explicitly. 

I have also added some steps to help with troubleshooting in case the user made some mistakes when building Mathesar from the source

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [ ] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by pavish** [comment]
@silentninja can you stack this on top of https://github.com/centerofci/mathesar/pull/2907. I think it might be better to get that one merged first and resolve conflicts here.

**Comment by pavish** [comment]
Blocked until https://github.com/centerofci/mathesar/pull/2907 is merged.

@silentninja Please resolve the conflicts.

File: docs/docs/installation/build-from-source/index.md [hunk 1]
```diff
@@ -31,6 +31,10 @@ You'll need to install the following system packages before you install Mathesar
 
     _(This is required for installation only and will eventually be [relaxed](https://github.com/centerofci/mathesar/issues/2871))_
 
+- [NPM](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) 14 or newer (Verify with `node --version`)
+
+    _(This is required for installation only and will eventually be [relaxed](https://github.com/centerofci/mathesar/issues/2871))_
```
Comment (pavish) [inline docs/docs/installation/build-from-source/index.md:36]: This one specifies npm but mentions the node version.

npm version required is 6. It is checked using `npm --version`.

**Review by pavish — CHANGES_REQUESTED** [review]

File: docs/docs/installation/build-from-source/index.md (hunk 1 above)
Comment (silentninja) [reply docs/docs/installation/build-from-source/index.md]: Sorry, fixed it.

**Review by pavish — APPROVED** [review]
@silentninja I added a commit https://github.com/centerofci/mathesar/pull/2924/commits/8f8700d8e21f4c99fc162894f37c3956bb6c7822 because `npm` is not capitalized. It's brand name is always lowercased.

**Pull request merged**